test/utils/wasm_engine.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "types.hpp"
+#include <memory>
+#include <optional>
+
+namespace fizzy
+{
+struct execution_result2
+{
+    bool trapped = false;
+    std::optional<uint64_t> value;
+};
+
+class WasmEngine
```

Comment:
Needs virtual destructor

Reply:
I feel a bit uneasy that it actually encapsulates one instance, but there's no indication of it in the name...

Reply:
Is `TestInstance` a better name?

Reply:
I think it's ok

Reply:
I'm not convinced though.